commonknowledge/meep-intelligence-hub
# Polling station data

This PR adds electoral commission polling station data.

## Description

- [x] Adds Electoral Commission data to the UI for mapping, and to the public enrichment API
- [x] Creates a reusable way to call third party APIs as part of the Updater pipeline
- [x] Reimplements the Updater data picker based on @Arbyhisenaj's latest Figma designs, loading the options from the API

## Motivation and Context

Some data is too dynamic, too big, or too complex to store locally in our own systems. For these cases we now have the ability to pull in data directly from third party APIs, alongside locally stored `DataSets` from imports and orgs' own data sources.

In this case, we are plugging in to Democracy Club / Electoral Commission's API so that campaigning organisations can get quick access to their nearest polling station, which will help with Get Out The Vote (GOTV) operations on voting day.

## How Can It Be Tested?

- Run the automated tests
- Try syncing electoral commission data to your CRM via the UI
- Check out the API

## How Will This Be Deployed?

- [ ] Add `ELECTORAL_COMMISSION_API_KEY` to env

## Screenshots (if appropriate):

<img width="1136" alt="Screenshot 2024-04-22 at 14 10 31" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/2dfce32f-a5ea-46a6-b3d0-6f2452608291">

<img width="1155" alt="Screenshot 2024-04-22 at 14 45 54" src="https://github.com/commonknowledge/meep-intelligence-hub/assets/237556/a27a7e98-b095-48c7-953d-3fc7ea5b3726">

## Types of changes
